torch/csrc/jit/passes/lower_tuples.cpp

```diff
@@ -28,18 +68,16 @@ static void VisitNode(Node* n, Node* insert_point) {
     return;
   }
 
-  // make any TupleUnpack dead by undoing TupleUnpack(TupleConstruct())
-  if(n->kind() == prim::TupleUnpack) {
-    auto construct = n->input()->node();
-    // note: removing these asserts changes this pass from a complete lowering
-    // pass to one that removes tuples when possible. When tuples are first-class
-    // in the interpreter, we should still run this pass to remove extraneous uses
-    JIT_ASSERTM(construct->kind() == prim::TupleConstruct, "tuple unpack not matched to tuple construct");
-    for(size_t i = 0; i < n->outputs().size(); ++i) {
-      n->outputs()[i]->replaceAllUsesWith(construct->inputs()[i]);
-    }
-    return;
+  // note: changing the second argument to false changes this pass from a complete lowering
+  // pass to one that removes tuples when possible. When tuples are first-class
+  // in the interpreter, we should still run this pass to remove extraneous uses
+
+  if(n->kind() == prim::TupleUnpack || n->kind() == prim::TupleIndex ||
```

Comment:
This guard is repeated at the beginning of `removeTuplesNodes`. So isn't this check redundant?

Reply:
The guard is for returning after removeTupleNodes is called